lib/rubocop/cop/style/combinable_defined.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Checks for multiple `defined?` calls joined by `&&` that can be combined
+      # into a single `defined?`.
+      #
+      # When checking that a nested constant or chained method is defined, it is
+      # not necessary to check each ancestor or component of the chain.
+      #
+      # @example
+      #   # bad
+      #   defined?(Foo) && defined?(Foo::Bar) && defined?(Foo::Bar::Baz)
+      #
+      #   # good
+      #   defined?(Foo::Bar::Baz)
+      #
+      #   # bad
+      #   defined?(foo) && defined?(foo.bar) && defined?(foo.bar.baz)
+      #
+      #   # good
+      #   defined?(foo.bar.baz)
+      class CombinableDefined < Base
+        extend AutoCorrector
+        include RangeHelp
+
+        MSG = 'Combine nested `defined?` calls.'
+
+        def on_and(node)
+          subjects = defined_subjects(node)
+          namespaces = namespaces(subjects)
+
+          subjects.each do |subject|
+            next unless namespaces.any?(subject)
+
+            add_offense(node) do |corrector|
+              remove_clause(corrector, subject)
+            end
+          end
+        end
+
+        private
+
+        def defined_subjects(node)
+          node.each_descendant(:defined?).filter_map do |defined_node|
+            next unless defined_node.parent&.and_type?
+
+            subject = defined_node.first_argument
+            subject if subject.const_type? || subject.call_type?
+          end
+        end
+
+        def namespaces(nodes)
+          nodes.filter_map do |node|
+            if node.respond_to?(:namespace)
+              node.namespace
+            elsif node.respond_to?(:receiver)
+              node.receiver
+            end
+          end
+        end
+
+        def remove_clause(corrector, clause)
+          # Remove the redundant `defined?` node, as well as the subsequent
+          # `&&` operator.
+
+          source = @processed_source.buffer.source
```

Comment:
It seems my explanation wasn’t clear. If describing the behavior of the entire method, it would be better to place it as a method comment. For example, something like this.

```suggestion
        # Remove the redundant `defined?` node, as well as the subsequent `&&` operator.
        def remove_clause(corrector, clause)
          source = @processed_source.buffer.source
```